### PR TITLE
book suggestion route implemented

### PR DESCRIPTION
**Pull request by jhonmunoz**
`create` route for `BookSuggestion` model implemented:
- non registered users allowed.
- validations implemented

### CODE REVIEW

File: app/controllers/api/v1/book_suggestions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class BookSuggestionsController < ApiController
+      skip_before_action :authenticate_request
+      def create
+        @suggestion = BookSuggestion.new(suggest_params)
+        @suggestion.user_id = current_user.id if current_user
```
Comment (hdf1996) [inline app/controllers/api/v1/book_suggestions_controller.rb]: You can do @suggestion.user = current_user, if the user is nil. the suggestion user will be nil too. Rails does the magic for the association

File: app/controllers/api/v1/book_suggestions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+        if @suggestion.valid?
```
Comment (hdf1996) [inline app/controllers/api/v1/book_suggestions_controller.rb]: you can do @suggestion.save directly, if it's invalid it wouldn't save anything

**Review by hdf1996 — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/book_suggestions_controller.rb (hunk 1 above)
Comment (jhonmunoz) [reply app/controllers/api/v1/book_suggestions_controller.rb]: gotcha

File: app/controllers/api/v1/book_suggestions_controller.rb (hunk 2 above)
Comment (jhonmunoz) [reply app/controllers/api/v1/book_suggestions_controller.rb]: Understood. Notice that I'll need to render an error if the request is invalid, so I still need to call for validation

**Review by hdf1996 — APPROVED** [review]

**Pull request merged**